homeassistant/exceptions.py

```diff
@@ -64,6 +64,13 @@ def __str__(self) -> str:
             return self._message
 
         if not self.generate_message:
+            # Initialize self._message to the string repr of the class
+            # to prevent a recursive loop.
+            self._message = str(self.__class__)
```

Comment:
```suggestion
            self._message = f"Parent class {self.__class__.__name__} is missing __str__ method."
```

I'm not sure its better, but its not clear what happened if consumer of the string ends up with `"<class 'tests.test_exceptions.test_home_assistant_error_subclass.<locals>._SubExceptionConstructor'>"` as the string value